diagnostics-app/src/app/core/services/events.service.ts

```diff
@@ -106,37 +102,44 @@ export class EventsService {
           this.subjects.set(category, new Subject<EventsInfo>);
         }
       } catch (err) {
-        return { success: false, error: String(err) };
+        return {success: false, error: String(err)};
       }
-      
     }
-    return { success: true, subject: this.subjects.get(category)! };
+    return {success: true, subject: this.subjects.get(category)!};
   }
 
-  isEventSupported: (eventType: EventCategory) => Promise<EventsResponse> = eventType => {
-    const payload: EventsRequest = {
-      action: EventsAction.IS_EVENT_SUPPORTED,
-      eventType,
-    };
-    const request = this.constructEventsRequest(payload);
-    return this.sendRequest(request);
-  }
+  isEventSupported: (eventType: EventCategory) => Promise<EventsResponse> =
+      eventType => {
+        const payload: EventsRequest = {
+          action: EventsAction.IS_EVENT_SUPPORTED,
+          eventType,
+        };
+        if (this.supportabilityCache.has(eventType)) {
+          return this.supportabilityCache.get(eventType)!;
+        }
+        const request = this.constructEventsRequest(payload);
+        const promise = this.sendRequest(request);
+        this.supportabilityCache.set(eventType, promise);
+        return promise;
+      }
 
-  startCapturingEvents: (eventType: EventCategory) => Promise<EventsResponse> = eventType => {
-    const payload: EventsRequest = {
-      action: EventsAction.START_CAPTURING_EVENT,
-      eventType,
-    };
-    const request = this.constructEventsRequest(payload);
-    return this.sendRequest(request);
-  }
+  startCapturingEvents: (eventType: EventCategory) => Promise<EventsResponse> =
+      eventType => {
+        const payload: EventsRequest = {
+          action: EventsAction.START_CAPTURING_EVENT,
+          eventType,
+        };
+        const request = this.constructEventsRequest(payload);
+        return this.sendRequest(request);
+      }
 
-  stopCapturingEvents: (eventType: EventCategory) => Promise<EventsResponse> = eventType => {
-    const payload: EventsRequest = {
-      action: EventsAction.STOP_CAPTURING_EVENT,
-      eventType,
-    };
-    const request = this.constructEventsRequest(payload);
-    return this.sendRequest(request);
-  }
+  stopCapturingEvents:
+      (eventType: EventCategory) => Promise<EventsResponse> = eventType => {
+        const payload: EventsRequest = {
+          action: EventsAction.STOP_CAPTURING_EVENT,
+          eventType,
+        };
+        const request = this.constructEventsRequest(payload);
+        return this.sendRequest(request);
+      }
```

Comment:
Suggestion: It seems that this part is updated by formatter. (Ditto to the import headers)

Maybe we should have a unified way to format these typescript files.

I think we need to move this kind of change into another PR if we want to use different formatter. Otherwise this PR will have a lot of unnecessary changes, which make it hard to notice the logic change.

Thanks!

Reply:
Done.

You are correct, I've opened up a bug b/308944438 and outlined my approach there.

I'm planning to run formatter on all files after this last PR is merged.